src/backend/access/appendonly/aosegfiles.c

```diff
@@ -1680,7 +1680,7 @@ get_ao_distribution(PG_FUNCTION_ARGS)
 	funcctx = SRF_PERCALL_SETUP();
 
 	query_block = (QueryInfo *) funcctx->user_fctx;
-	if (query_block->index < query_block->rows)
+	if (query_block != NULL && query_block->index < query_block->rows)
```

Comment:
I don't think we should tolerate NULL query_block. I suppose it's better to add an assertion

Reply:
No! funcctx->user_fctx is assigned not always